src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ConfigurationTests.cs

```diff
@@ -202,14 +202,62 @@ public void FullConfigurationTest(bool redact)
             builder.ReplaceDotnetEnvironment();
             builder.ReplaceMonitorEnvironment();
 
-            // Build the host and get the Urls property from configuration.
+            // Build the host and get the configuration.
             IHost host = builder.Build();
             IConfiguration rootConfiguration = host.Services.GetRequiredService<IConfiguration>();
 
+            string generatedConfig = WriteAndRetrieveConfiguration(rootConfiguration, redact);
+
+            Assert.Equal(CleanWhitespace(generatedConfig), CleanWhitespace(ConstructExpectedOutput(redact)));
+        }
+
+        /// <summary>
+        /// Tests that the connection mode is set correctly for various configurations of the diagnostic port
+        /// </summary>
+        [Theory]
+        [MemberData(nameof(DiagnosticPortTestsHelper.GetFileNamesAndEnvironmentVariables), MemberType = typeof(DiagnosticPortTestsHelper))]
```

Comment:
If the GetFileNamesAndEnvironmentVariables method is moved to this class, then you can just refer to it as `[MemberData(nameof(DiagnosticPortTestsHelper.GetFileNamesAndEnvironmentVariables))]` without the member type qualification.